src/Data/List/Relation/Unary/All/Properties.agda

```diff
@@ -466,23 +472,20 @@ take⁺ zero    pxs        = []
 take⁺ (suc n) []         = []
 take⁺ (suc n) (px ∷ pxs) = px ∷ take⁺ n pxs
 
-takeWhile⁺ : (Q? : Decidable Q) → All P xs → All P (takeWhile Q? xs)
-takeWhile⁺               Q? []         = []
-takeWhile⁺ {xs = x ∷ xs} Q? (px ∷ pxs) with does (Q? x)
-... | true  = px ∷ takeWhile⁺ Q? pxs
-... | false = []
+takeWhileP⇒Q⇒R⁺ : (∀ {x} → P x → Q x → R x) → (Q? : Decidable Q) →
+                  All P xs → All R (takeWhile Q? xs)
```

Comment:
Isn't this a a combination of `All Q (takeWhile Q? xs)`, the old `takeWhile⁺`  and `zipWith`?
It feels like we're dropping a separation of concerns in favour of a swiss knife approach.

Reply:
It does rather feel like a composition of things, doesn't it?

Reply:
Thanks very much for these observations: I hadn't realised how much of a Swiss-Army-type person I can be! Very helpful to have it pointed out. I think this is further support for an abandon.